### PR TITLE
Upgrade dependencies to new versions

### PR DESCRIPTION
**Pull request by sarathms**
Target dependencies: 
```
"rebass": "2.3.3",
"@rebass/grid": "6.0.0",
"styled-components": "4.0.0-beta.3",
"styled-system": "3.1.3"
```

### CODE REVIEW

**Comment by sarathms** [comment]
There might be a few more commits coming up after checking if `styled-component`'s API changes have more impact in places I haven't looked yet.

File: components/globe.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ class WebGLGlobe extends React.Component {
 
     if (webGLEnabled) {
       return (
-        <StyledGlobe innerRef={(el) => { this.globeRef = el }} />
+        <StyledGlobe ref={(el) => { this.globeRef = el }} />
```
Comment (hellais) [inline components/globe.js]: It's cool that they have fixed this. Do you know if this requires a particular version of React (I seem to recall this wasn't possible in previous styled-components version due to react limitations)? 

Be sure to do some testing of this (especially when used in form/input components) as there may be some breakage.

File: components/layout.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ import Footer from './footer'
 
 theme.maxWidth = 1024
 
-injectGlobal`
+const GlobalStyle = createGlobalStyle`
```
Comment (hellais) [inline components/layout.js]: We should probably add this to design-system

File: components/globe.js (hunk 1 above)
Comment (sarathms) [reply components/globe.js]: This is based on the `forwardRef` feature coming in with [`react@16.3.0`](https://reactjs.org/blog/2018/03/29/react-v-16-3.html#forwardref-api). Which means we have to upgrade react too which seems to be a non-breaking upgrade. We should expedite writing tests.

**Pull request merged**